### DISCUSSION TITLE
Passing service to per-message closure

### DISCUSSION THREAD

**Discussion by maquefel** [protobuf-c/protobuf-c#660] [Q&A]
It would be nice to distinguish calling service in per-message closure (ProtobufCClosure). For what exactly closures are used except for protobuf-c-rpc? 

Is it possible to pass service pointer during closure call ?

**Comment by lipnitsk** [comment]
You should be able to put anything you want into closure_data, and yes, these types are not used inside protobuf-c. Please file a ticket or a pull request in protobuf-c-rpc if needed.
